### PR TITLE
Improve build and push scripts.

### LINKED ISSUE #208: Improve the release process
For example, we need to add a tag to git, probably should automatically rev package.json and package-lock.json...

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes #208 

@drubin @itowlson

### CODE REVIEW

File: push-package.sh [hunk 1]
```diff
@@ -1,3 +1,12 @@
 #!/bin/bash
+if [[ "${1}x" == "x" ]]; then
+  echo "Usage: ${0} <version>"
```
Comment (drubin) [inline push-package.sh]: Should we not just pull the version out the package.json?

I am totally fine leaving this manual though.

File: push-package.sh [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+  exit 1
+fi
+
+. ./pre-check.sh
+
+git tag ${1}
+git push upstream ${tag}
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply push-package.sh]: fixed.

File: push-package.sh (hunk 1 above)
Comment (brendandburns) [reply push-package.sh]: fixed to pull it from `package.json` thanks for the idea!

**Comment by brendandburns** [comment]
@drubin comments addressed, please re-check.

Thanks!

**Comment by drubin** [comment]
@brendandburns it appears like you have forgotten to push your changes?

Also something I always do out of habit with bash is adding `#!/bin/bash -e` so if we ever end up using them in CI they fail early and we have proper error handling. 

Is there a reason to not do it?

**Comment by brendandburns** [comment]
oops, now pushed. *sigh*

And `set -e` added also.

**Comment by drubin** [comment]
@brendanburns before merging I just wanted us to maybe take a quick look at https://github.com/sindresorhus/np 

We can also merge this now and add another ticket to look at this after, but since you are the primary person making the releases I will leave this up to you.

**Comment by brendandburns** [comment]
@drubin cool, I'll check it out. I think in the meantime let's merge this since it's already done and its an improvement on what we currently have.

But `np` looks even better :)

**Pull request merged**